app/javascript/hyacinth_ui_v1/components/digital_objects/new/AssetNew.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Form } from 'react-bootstrap';
+import { useMutation } from '@apollo/react-hooks';
+import uploadFile from '../../../graphql/file_uploads';
+import { createAssetMutation } from '../../../graphql/assets';
+
+function debugObject(obj) {
+  console.log(obj);
+  return new Promise(resolve => resolve(obj));
+}
+
+function AssetNew(props) {
+  const { parentId, refetch } = props;
+  const [createAsset] = useMutation(createAssetMutation);
+
+  const onChange = (event) => {
+    const {
+      target: {
+        validity,
+        files: [file],
+      },
+    } = event;
+    if (!validity.valid) return false;
+    function resetFileInput() { document.getElementById(AssetNew.elementId).value = ''; }
+    const url = `/api/v1/digital_objects/${parentId}/uploads`;
+    return uploadFile(file, url)
+      .then(blobAttributes => createAsset(
+        {
+          variables: {
+            input: {
+              signedBlobId: blobAttributes.signed_id,
+              parentId,
+            },
+          },
+          update: refetch,
+        },
+      ))
+      .then(resetFileInput);
+  };
+  return (
+    <Form>
+      <input id={AssetNew.elementId} type="file" required onChange={onChange} parentid={parentId} direct_upload="true" />
+    </Form>
+  );
+}
+
+AssetNew.elementId = `input-${String.toString(Math.random()).replace('.', '')}`;
```

Comment:
I think this goes against good reusable component design, since you're effectively creating a static variable that will be the same for every occurrence of this component for the lifetime of the react app. Things like unique element id would be better off set in the component's state.

Reply:
Done.

Reply:
👍